### ISSUE TITLE
Unclear regression in DLNA server with LG WebOS TV in rclone version 1.59

### ISSUE THREAD

**Issue by fmagin** [rclone/rclone#6354]
#### What is the problem you are having with rclone?

I am using rclone as a DLNA server for my LG TV, and since rclone version 1.59 the TV only reports "No files exist".

I bisected this down to 59c242bbf6c3dac3cff848691f84f58bd69a29cf i.e. this is the first commit that results in this error. Sadly this is a dependency update commit which makes it a lot harder to judge what change caused this.
VLC still finds the files via the server, so this is some weird interaction between something that changed in this commit and how LG WebOS expects a DLNA server to behave. I don't see a dependency that contains `DLNA` in the name, but if someone tells me likely candidates that could have caused this I can check those too, and then open an issue with that project if applicable.

#### What is your rclone version (output from `rclone version`)

This is with the release from Github
```
$ rclone-v1.59.0-linux-amd64/rclone version
rclone v1.59.0
- os/version: arch "rolling" (64 bit)
- os/kernel: 5.18.14-arch1-1 (x86_64)
- os/type: linux
- os/arch: amd64
- go/version: go1.18.3
- go/linking: static
- go/tags: none
```

#### Which OS you are using and how many bits (e.g. Windows 7, 64 bit)

arch "rolling" (64 bit)

#### Which cloud storage system are you using? (e.g. Google Drive)

none

#### The command you were trying to run (e.g. `rclone copy /tmp remote:tmp`)

`rclone serve dlna /storage_ssd/video/`

#### A log from the command with the `-vv` flag (e.g. output from `rclone -vv copy /tmp remote:tmp`)

```
2022/08/01 22:04:02 DEBUG : rclone: Version "v1.60.0-beta.6376.3a8e52de7" starting with parameters ["rclone-v1.60.0-beta.6376.3a8e52de7-linux-amd64/rclone" "serve" "dlna" "--addr" ":7880" "--name" "monolith" "/storage_ssd/video/" "-vvv"]
2022/08/01 22:04:02 DEBUG : Creating backend with remote "/storage_ssd/video/"
2022/08/01 22:04:02 DEBUG : Using config file from "/home/fmagin/.config/rclone/rclone.conf"
2022/08/01 22:04:02 INFO  : Local file system at /storage_ssd/video/: poll-interval is not supported by this remote
2022/08/01 22:04:02 NOTICE: Local file system at /storage_ssd/video/: Serving HTTP on [::]:7880
2022/08/01 22:04:02 INFO  : DLNA server on :7880: Started SSDP on virbr2
2022/08/01 22:04:02 INFO  : DLNA server on :7880: Started SSDP on virbr1
2022/08/01 22:04:02 INFO  : DLNA server on :7880: Started SSDP on enp34s0
2022/08/01 22:04:02 INFO  : DLNA server on :7880: Started SSDP on virbr0
2022/08/01 22:04:03 INFO  : /static/ContentDirectory.xml: 192.168.178.39:60636 GET 200  
2022/08/01 22:04:03 INFO  : /static/ContentDirectory.xml: 192.168.178.39:60634 GET 200  
2022/08/01 22:04:03 INFO  : /static/ConnectionManager.xml: 192.168.178.39:60634 GET 200  
2022/08/01 22:04:03 INFO  : /static/ConnectionManager.xml: 192.168.178.39:60636 GET 200  
2022/08/01 22:04:03 INFO  : /static/X_MS_MediaReceiverRegistrar.xml: 192.168.178.39:60634 GET 200  
2022/08/01 22:04:03 INFO  : /static/X_MS_MediaReceiverRegistrar.xml: 192.168.178.39:60636 GET 200  
2022/08/01 22:04:03 ERROR : /: 192.168.178.39:60634 SUBSCRIBE 404  
2022/08/01 22:04:03 ERROR : /: 192.168.178.39:60636 SUBSCRIBE 404  
2022/08/01 22:04:03 ERROR : /: 192.168.178.39:60638 SUBSCRIBE 404  
2022/08/01 22:04:03 ERROR : /: 192.168.178.39:60640 SUBSCRIBE 404  
2022/08/01 22:04:03 INFO  : /ctl: 192.168.178.39:60642 POST 200 "urn:schemas-upnp-org:service:ContentDirectory:1#GetSearchCapabilities" 
2022/08/01 22:04:04 INFO  : /ctl: 192.168.178.39:60642 POST 200 "urn:schemas-upnp-org:service:ContentDirectory:1#Browse" 
2022/08/01 22:04:37 INFO  : /ctl: 192.168.178.39:60642 POST 200 "urn:schemas-upnp-org:service:ContentDirectory:1#Browse" 
```
It looks the same with a working version, i.e. those `SUBSCRIBE 404  ` errors also show up when it's working.

#### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

**Comment by jorams** [comment]
DLNA also stopped working with my LG TV as of 1.59. When opening the server over DLNA, I get the following error:

> Your device is disconnected. Please check the connection status.

The log looks the same.

Looking at the updates in 59c242bbf6c3dac3cff848691f84f58bd69a29cf, the one most likely to cause this is the upgrade of [anacrolix/dms](https://github.com/anacrolix/dms) from 1.3.0 to 1.4.0. That's these 13 commits: https://github.com/anacrolix/dms/compare/v1.3.0...v1.4.0. It includes https://github.com/anacrolix/dms/pull/86, which explicitly mentions fixing DLNA browsing for LG TVs.

I plan to do some actual testing later this weekend.

It's worth noting that [anacrolix/dms](https://github.com/anacrolix/dms) is now at 1.5.0, which is included in master as of 2 hours ago (df513ca90a9dfff9f7dca9cc38e6ddc832afb597).

**Comment by ncw** [comment]
Great detective work.

Does the master branch fix it?

**Comment by jorams** [comment]
Sadly it does not.

I just tried builds of master with different versions of anacrolix/dms.

- anacrolix/dms@v1.5.0 has the problem
- anacrolix/dms@v1.4.0 has the problem
- anacrolix/dms@v1.3.0 does not have the problem
- anacrolix/dms@3d402bc95f0bde7d353e23cf4f3c5e4fdc37c9ec, the squashed PR I suspected above, has the problem
- anacrolix/dms@980b2c66b729b714cc013688b62f0911cb3225f7, the commit before the PR, does not have the problem

So it looks like something in that PR breaks it.
